pkg/test/syncset/syncset.go

```diff
@@ -116,6 +117,20 @@ func WithResources(objs ...hivev1.MetaRuntimeObject) Option {
 	}
 }
 
+// Each resource is a separate string. Don't pass in one string with a yaml list.
+func WithYAMLResources(objs ...string) Option {
+	return func(syncSet *hivev1.SyncSet) {
+		syncSet.Spec.Resources = make([]runtime.RawExtension, len(objs))
+		for i, obj := range objs {
+			jsonObj, err := yaml.YAMLToJSON([]byte(obj))
```

Comment:
A comment about ensuring the provided YAML resources are JSON compatible (no null keys in your maps for ex,) would be a good idea.

Reply:
I thought yaml was a superset of json. Is it even possible to express a null map key in yaml?

Reply:
Added a comment. It seems kind of obvious, but no harm.

Reply:
I got this from [here](https://pkg.go.dev/sigs.k8s.io/yaml#YAMLToJSON)
> In YAML you can have binary and null keys in your maps. These are invalid in JSON, and therefore int, bool and float keys are converted to strings implicitly.

Reply:
Ack